api/src/pcapi/core/bookings/repository.py

```diff
@@ -906,3 +908,25 @@ def _serialize_csv_report(query: Query) -> str:
         )
 
     return output.getvalue()
+
+
+def get_unretrieved_bookings(expiration_days_delta: int) -> typing.Generator[Booking, None, None]:
+    """
+    Find unretrieved bookings that will expire in exactly
+    `expiration_days_delta` days.
+    """
+    query = (
+        Booking.query.options(
+            joinedload(Booking.stock).load_only(Stock.id).joinedload(Stock.offer).load_only(Offer.subcategoryId)
+        )
+        .join(Stock, Offer)
+        .filter_by(canExpire=True)
+        .filter(Booking.status.in_([BookingStatus.PENDING, BookingStatus.CONFIRMED]))
```

Comment:
Ça ne devrait concerner que les offres individuelles non ?
En ce cas, je pense qu'on peut faire deux choses, 
- filtrer sur individualBookingId not None ou passer par IndividualBooking.query
- Pas besoin de passer sur le statut PENDING